models/AddressLocation.yaml

```diff
@@ -0,0 +1,71 @@
+type: object
+name: AddressLocation
+x-stoplight:
+  id: s3q6yalatyags
+description: "\t\nAn interface used to express a location using a `UnLocationCode`."
+properties:
+  locationName:
+    type: string
+    format: int64
+    description: The name of the location.
+    pattern: ^\S+(\s+\S+)*$
+    example: Port of Amsterdam
+    maxLength: 100
+  locationType:
+    type: string
+    enum:
+      - ADT
+    description: Discriminator used to identify this as a `UNLocation` location interface
+    example: ADT
+  address:
+    type: object
+    description: "\t\nAn object for storing address related information."
+    properties:
+      name:
+        type: string
+        pattern: ^\S+(\s+\S+)*$
+        maxLength: 100
+        example: John
+        description: Name of the address.
+      street:
+        type: string
+        maxLength: 100
+        description: The name of the street of the party’s address.
+        example: Kronprinsessegade
+      streetNumber:
+        type: string
+        maxLength: 50
+        example: '54'
+        description: The number of the street of the party’s address.
+      floor:
+        type: string
+        pattern: ^\S+(\s+\S+)*$
+        maxLength: 50
+        example: 5. sal
+        description: The floor of the party’s street number.
+      postCode:
+        type: string
+        description: The post code of the party’s address.
+        example: '1306'
+        maxLength: 50
+      city:
+        type: string
+        pattern: ^\S+(\s+\S+)*$
+        maxLength: 65
+        example: København
+        description: The city name of the party’s address.
+      stateRegion:
+        type: string
+        description: The state/region of the party’s address.
+        maxLength: 65
+        example: Copenhagen
+      country:
+        type: string
+        pattern: ^\S+(\s+\S+)*$
+        example: Denmark
+        maxLength: 75
+        description: The country of the party’s address.
+    required:
+      - name
+required:
+  - locationType
```

Comment:
I think we should make the address object a required field - otherwise it makes no sense

Reply:
Agreed